core/templates/dev/head/components/ck-editor-helpers/ck-editor-4-rte.directive.ts

```diff
@@ -86,8 +89,10 @@ angular.module('oppia').directive('ckEditor4Rte', [
         }).join(',');
         var buttonNames = [];
         names.forEach(function(name) {
-          buttonNames.push('Oppia' + name);
-          buttonNames.push('-');
+          if (!(contextIsExplorationPlayer)) {
```

Comment:
If I understand the code correctly -- suppose contextIsExplorationPlayer is true. Wouldn't the code above already cause `names` to be an empty array, such that you don't need this "if" condition here?

Reply:
No @seanlip, When the `contextIsExplorationPlayer` is True `names` will have all the components. (note that the `!` is not a part of the contextIsExplorationPlayer statement, the effect of  `!` ends before we check  `contextIsExplorationPlayer`)

Reply:
We need the `names` to have all the components in order for enverything to get rendered in ckeditor